### PR TITLE
fix(email validation): make email validation case insensitive

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/108

#### List of Changes
Make mail validation case insensitive

#### Motivation and Context
Make mail validation case insensitive

#### How Has This Been Tested?
Junit tests + local test with ecommerce local

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/pagopa/transactions/controllers/TransactionsControllerTest.java [hunk 1]
```diff
@@ -657,6 +659,59 @@ void shouldGetTransactionInfoInAllStatuses() {
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"foo@test.it", "FoO@TeSt.iT", "FOO@TEST.IT"})
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/controllers/TransactionsControllerTest.java:663]: immagine

**Review by giovanniberti — APPROVED** [review]

File: api-spec/transactions-api.yaml [hunk 2]
```diff
@@ -356,7 +356,7 @@ components:
         email:
           description: Email string format
           type: string
-          pattern: (?:[a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+\/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?:25[0-5]|2[0-4]\d|[01]?\d\d?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])
+          pattern: (?i)(?:[a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+\/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?:25[0-5]|2[0-4]\d|[01]?\d\d?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])
```
Comment (giovanniberti) [inline api-spec/transactions-api.yaml:359]: Are you sure this syntax is correct?

File: src/test/java/it/pagopa/transactions/controllers/TransactionsControllerTest.java (hunk 1 above)
Comment (pietro-tota) [reply src/test/java/it/pagopa/transactions/controllers/TransactionsControllerTest.java:663]: Ahahahhahahahha 🤣

File: api-spec/transactions-api.yaml (hunk 2 above)
Comment (pietro-tota) [reply api-spec/transactions-api.yaml]: Changed the regex removing the case insensitive mode modifying the regex

**Review by Fernando-Granato — APPROVED** [review]
Good job! 🚀

**Pull request merged**